lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2137,9 +2369,74 @@ namespace {
             ConcurrentReferenceKind::CrossActor);
       }
 
+      case ActorIsolationRestriction::DistributedActorSelf: {
```

Comment:
Hmm. This is crying out for a refactor of isolation checking. I won't hold you to doing that before landing this, but I expect we can simplify this block considerably.

Reply:
Yes that sounds true, I'm happy to clean up the isolation checking in an isolated (pun intended) PR.